java/ingest/src/main/java/sleeper/ingest/job/IngestJobRunner.java

```diff
@@ -116,26 +116,24 @@ public IngestResult ingest(IngestJob job) throws InterruptedException, IteratorE
                         + ". This file will be ignored and will not be ingested.", pathString);
             }
         }
+        // Run the ingest
+        try (CloseableIterator<Record> recordIterator = new ConcatenatingIterator(inputIterators);
+             BufferAllocator bufferAllocator = new RootAllocator()) {
 
-        // Concatenate iterators into one iterator
-        CloseableIterator<Record> concatenatingIterator = new ConcatenatingIterator(inputIterators);
-
-        // Get StateStore for this table
-        StateStore stateStore = stateStoreProvider.getStateStore(tableProperties);
+            IngestCoordinatorFactory factory = IngestCoordinatorFactory.builder()
```

Comment:
The name IngestCoordinatorFactory suggests that it should be reused. Can we find a way to keep the BufferAllocator out of the factory object, so that it can be reused?

Reply:
We could make this a separate issue if that helps.

Reply:
I've made it so the bufferAllocator is created inside the factory class, as part of the work related to the comment about performing the full ingest in the factory